tests/ignite/engine/test_create_supervised.py

```diff
@@ -340,25 +369,39 @@ def _test_create_evaluation_step(
     assert output_transform_mock.called
 
 
-def test_create_supervised_trainer():
+@pytest.mark.parametrize("gradient_accumulation_steps", [1, 3])
```

Comment:
I do not understand why you changed this ?

Reply:
You mean there is no need to `_test_create_supervised_trainer_have_grad_after_iteration` with different `gradient_accumulation_steps`?

Reply:
@vfdev-5 You mean there is no need to _test_create_supervised_trainer_have_grad_after_iteration with different gradient_accumulation_steps?

Reply:
@sadra-barikbin try to execute these tests and see what this code do.
It will generate 2 tests like 
`test_create_supervised_trainer[gradient_accumulation_steps-1]` and `test_create_supervised_trainer[gradient_accumulation_steps-3]` where 
`_test_create_supervised_trainer_wrong_accumulation()` and `_test_create_mocked_supervised_trainer()` will be executed twice which is useless.

I suggest to revert the change and just call 
```
_test_create_supervised_trainer_have_grad_after_iteration(1)
_test_create_supervised_trainer_have_grad_after_iteration(3)
```

Another solution could be split `test_create_supervised_trainer` into 2 tests such that the first test is executing once `_test_create_supervised_trainer_wrong_accumulation` and `_test_create_mocked_supervised_trainer` and the second is parametrizing `gradient_accumulation_steps` and calling others

Reply:
> Another solution could be split test_create_supervised_trainer into 2 tests such that the first test is executing once _test_create_supervised_trainer_wrong_accumulation and _test_create_mocked_supervised_trainer and the second is parametrizing gradient_accumulation_steps and calling others

@sadra-barikbin can you make suggested changes please ?

Reply:
Excuse me for delay. Sure.

Reply:
@vfdev-5 , I did the first one but also did some refactors. Does it look good?

Reply:
It's difficult to say as this adds more lines than it removes :)
Can you pull from your branch locally and push an empty commit to restart the CI. Autofix pep does not launch everything somehow...

Reply:
It's because of applying autopep(It has put each parameter of `pytest.param`s in a separate line). Now there is no redundancy.

Reply:
@vfdev-5 , what else should I do?